src/spyglass/utils/dj_mixin.py

```diff
@@ -122,6 +127,65 @@ def fetch_nwb(self, *attrs, **kwargs):
         """
         return fetch_nwb(self, self._nwb_table_tuple, *attrs, **kwargs)
 
+    def fetch_pynapple(self, *attrs, **kwargs):
+        """Get a pynapple object from the given DataJoint query.
+
+        Parameters
+        ----------
+        query_expression : query
+            A DataJoint query expression (e.g., join, restrict) or a table to call fetch on.
+        nwb_master : tuple
+            Tuple (table, attr) to get the NWB filepath from.
+            i.e. absolute path to NWB file can be obtained by looking up attr column of table
+            table is usually Nwbfile or AnalysisNwbfile;
+            attr is usually 'nwb_file_abs_path' or 'analysis_file_abs_path'
+        *attrs : list
+            Attributes from normal DataJoint fetch call.
+        **kwargs : dict
+            Keyword arguments from normal DataJoint fetch call.
+
+        Returns
+        -------
+        nwb_objects : list
+            List of dicts containing fetch results and NWB objects.
+        """
+        if pynapple is None:
+            raise ImportError("Pynapple is not installed.")
+
+        query_expression, nwb_master = self, self._nwb_table_tuple
```

Comment:
I adjusted this syntax [here](https://github.com/CBroz1/spyglass/blob/log/src/spyglass/utils/dj_helper_fn.py#L136-L153) for #875. Probably worth carving out this piece of the `fetch_nwb` source into a separate `get_nwb_table` func that `fetch_pynapple` can import